### PR TITLE
Create a tmp directory if one doesn't exists or is read only #315

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/core/OS.java [hunk 1]
```diff
@@ -115,6 +115,19 @@
     private OS() {
     }
 
+    private static String findTmp() {
+        String target = System.getProperty("project.build.directory");
+        if (target != null)
+            return target + File.pathSeparator + "tmp";
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/OS.java:121]: should we create the directory in this case?

File: src/main/java/net/openhft/chronicle/core/OS.java [hunk 2]
```diff
@@ -115,6 +115,19 @@
...
+        final String tmp = System.getProperty("java.io.tmpdir");
+        if (tmp != null
+                && new File(tmp).isDirectory()
+                && new File(tmp).canWrite())
+            return tmp;
+        new File("tmp").mkdirs();
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/OS.java:127]: what if mkdirs() fails?

**Review by JerryShea — APPROVED** [review]

**Review by RobAustin — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/OS.java (hunk 2 above)
Comment (peter-lawrey) [reply src/main/java/net/openhft/chronicle/core/OS.java:127]: A subsequent IO operation that needs it will fail.

Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/OS.java:127]: sonar will warn about ignoring the return value - maybe we should at the very least assert the return value?

**Pull request merged**